homeassistant/components/media_player/onkyo_serial.py

```diff
@@ -0,0 +1,323 @@
+"""
+Support for Onkyo Receivers.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/media_player.onkyoserial/
+"""
+import logging
+import threading
+import re
+
+import voluptuous as vol
+
+from homeassistant.components.media_player import (
+    SUPPORT_TURN_OFF, SUPPORT_TURN_ON, SUPPORT_VOLUME_MUTE, SUPPORT_VOLUME_SET,
+    SUPPORT_SELECT_SOURCE, SUPPORT_PLAY, MediaPlayerDevice, PLATFORM_SCHEMA)
+from homeassistant.const import (STATE_OFF, STATE_ON, CONF_NAME)
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = []

```

Comment:
This isn't any different than denon.py, itunes.py. From what I can tell, other components that do not do this have external libraries available on pypi or github which I do not.

Is there another option since moving this and hosting it in a full fledged external library is beyond the scope of a simple hass controller.